src/openms/source/ANALYSIS/OPENSWATH/DIAScoring.cpp

```diff
@@ -201,17 +201,44 @@ namespace OpenMS
   }
 
   /// Precursor isotope scores
+  void DIAScoring::dia_ms1_isotope_scores(double precursor_mz, SpectrumPtrType spectrum,
+                                          double& isotope_corr, double& isotope_overlap, const EmpiricalFormula& sum_formula) const
+  {
+    //TODO precursor_mz can be received from the empirical formula (if required non-empty)
+
+    //collect the potential isotopes of this peak
+    int charge_state = sum_formula.getCharge();
+    double max_ratio;
+    int nr_occurrences;
+    std::vector<double> isotopes_int;
+    for (int iso = 0; iso <= dia_nr_isotopes_; ++iso)
+    {
+      double left  = precursor_mz + iso * C13C12_MASSDIFF_U / static_cast<double>(charge_state);
+      double right = left;
+      DIAHelpers::adjustExtractionWindow(right, left, dia_extract_window_, dia_extraction_ppm_);
```

Comment:
am I the only one who thinks this code is utterly non-intuitive?
I assumed it's broken, twice, when I saw it. But it seems to do the right thing, if you follow all the clues.
1) why pass two doubles by ref (which need to have the same value), instead of passing one and returning 2?
   (good place for using structured binding :))
2) switching the order of parameters (right, left) instead of (left, right), ... wow... just wow.

Reply:
haha true, I was distracted by the actually wrong stuff around here.